.pre-commit-config.yaml

```diff
@@ -0,0 +1,64 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+    # General
+    -   id: check-merge-conflict
+        fail_fast: true
+    -   id: check-case-conflict
+        fail_fast: true
+    -   id: check-added-large-files
+    -   id: check-symlinks
+    -   id: destroyed-symlinks
+    # Syntax
+    -   id: check-json
+    -   id: check-yaml
+    # Whitespace
+    -   id: mixed-line-ending
+    -   id: trailing-whitespace
+
+-   repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+    # Python
+    -   id: python-use-type-annotations
+    -   id: python-no-log-warn
+
+    # Documentation
+    -   id: rst-backticks
+    -   id: rst-directive-colons
+    -   id: rst-inline-touching-normal
+
+-   repo: https://github.com/abravalheri/validate-pyproject
+    rev: v0.10.1
+    hooks:
+    - id: validate-pyproject
+
+-   repo: https://github.com/asottile/setup-cfg-fmt
+    rev: v2.2.0
+    hooks:
+    -   id: setup-cfg-fmt
+
+-   repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.210
+    hooks:
+    - id: ruff
+
+#
+# Slow
```

Comment:
How slow?

Reply:
About a second or less. They are commented out because `1 second` is slow enough to be annoying for a pre-commit hook; but they are left in because that way they can be used by someone as desired.

Reply:
You might be inclined to enable these for CI.

Reply:
How do I switch this on automatically for CI without editing the source? Is 1 second really that bad?

Reply:
Sorry, I meant each commented out check is <= 1 second; collectively they sum. For CI, you'd edit source.

I think that since these are commented out by default, it's ok to add this section to CI later rather than immediately if that's a big issue. I.e. since these are commented out, we can make them suggested rather than enforced, as users with pre-commit wouldn't use these either unless they edited this file.

Though, _ultimately_, I do think we should add these in. Some are very useful. `check-ast` ensures your python parses, while `debug-statements` helps people not accidentally leave `import ipdb; ipdb.set_trace()` or `breakpoint()` in their code; which I just made PRs removing a few of

Reply:
Probably the right way to do CI is to have a second pre-commit config file and have the CI script be:

```bash
pre-commit run --all-files
precommit run --all-files --config .pre-commit-config-slow.yaml
```

Reply:
I can split this file into two if that's what is wanted @twizmwazin

Reply:
Ping @twizmwazin